File: src/model/user/ImgAddQO.ts
import DomFile from '@/model/DomFile'

export default class ImgAddQO {
  //自己拼出来的
  public src: string = null
  //小程序只有path和size
  size: number = null
  //aspectRatio 自己计算出来的比例
  aspectRatio: number = null
  //quality 自己计算出来的压缩比
  quality: number = null

  constructor (domFile: DomFile) {
    this.src = domFile.src
    this.size = domFile.size
    this.aspectRatio = domFile.aspectRatio
    this.quality = domFile.quality
  }
}

File: social-sdk/src/main/java/com/socialuni/sdk/model/QO/ImgAddQO.java
package com.socialuni.sdk.model.QO;


import com.socialuni.sdk.constant.AppConfigConst;
import com.socialuni.sdk.constant.ContentType;
import com.socialuni.sdk.constant.status.ContentStatus;
import com.socialuni.sdk.model.DO.user.UserDO;
import com.socialuni.sdk.model.DO.user.UserImgDO;
import lombok.Data;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.util.Date;

@Data
public class ImgAddQO {
    @NotBlank
    private String src;
    @NotNull
    private Double aspectRatio;
    //压缩率
    private Double quality;
    private Integer size;

    public UserImgDO toUserImgDO(UserDO user, String imgUrl) {
        //这里需要记录，变更历史，通过照片有效无效记录，
        UserImgDO userImgDO = new UserImgDO();
        userImgDO.setSrc(imgUrl + this.getSrc());
        userImgDO.setAspectRatio(this.getAspectRatio());
        userImgDO.setQuality(this.getQuality());
        userImgDO.setSize(this.size);
        userImgDO.setUserId(user.getId());
        userImgDO.setStatus(ContentStatus.enable);
        userImgDO.setCreateTime(new Date());
        userImgDO.setContent(AppConfigConst.img_content);
        userImgDO.setReportContentType(ContentType.userImg);
        userImgDO.setReportNum(0);
        userImgDO.setIsSelfAuth(false);
        user.setAvatar(userImgDO.getSrc() + "!avatar");
        return userImgDO;
    }
}
